design/welcome-email.feature

```diff
@@ -0,0 +1,93 @@
+Feature: Welcome new subscribers
+  - As a store owner
+  - I want to send welcome emails
+  - So that I will connect with my new subscribers immediately
+
+  Entities:
+    Guest - A person who has created an order or signed up for a newsletter. Guest has not created
+      unique customer account, but has a distinct email address in Magento store.
+    Customer - A person who has created an account in store.
+    Newsletter Subscriber - Person who exists in newsletter subscribers list in Magento. Newsletter
+      subscriber can have subscribed and unsubscribed status.
+    Need to Confirm - A setting in Magneto store that requires subscribers to confirm their
+      subscription before they can receive newsletters.
+
+  Background:
+    Given store owner has created opt-in automation in Smaily
+
+  Rule: Welcome email is sent to new subscribers
+    Scenario: Guest subscribes
+      Given Guest fills newsletter sign-up form
+      When guest submits the form
+      Then welcome email is sent
```

Comment:
It is not guaranteed that welcome email is sent, because email address might already exist in Smaily (Magento store is not connected to fresh Smaily organization).

Reply:
Should I introduce the Smaily logic here too? Currently, I think the idea was that this module (or this description) should not be aware of the logic that Smaily is doing. It just tries to send a welcome email. Maybe the wording of the result can be improved, but I can only come up with too technical text

> Then welcome email logic is activated in Smaily

Reply:
I think we should avoid trying to describe Smaily logic here. Stating that `welcome email is sent` is totally fine, because that needs to happen.

But, if we think in the context of the extension - can we even describe the welcome email feature? I mean if we would build it as a standalone feature (without having any knowledge about customer synchronization; or without having it enabled) how can we know if Guest, Newsletter Subscriber or Customer exists in Smaily? Do we need to check it before sending the API request to create subscriber in Smaily?